sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -102,7 +102,8 @@ case class InMemoryTableScanExec(
     case IsNull(a: Attribute) => statsFor(a).nullCount > 0
     case IsNotNull(a: Attribute) => statsFor(a).count - statsFor(a).nullCount > 0
 
-    case In(_: AttributeReference, list: Seq[Expression]) if list.isEmpty => Literal.FalseLiteral
+    // We rely on the optimizations in org.apache.spark.sql.catalyst.optimizer.OptimizeIn
```

Comment:
We should not rely on Optimizer for fixing the bugs. 

We need to fix the line 107 anyway.